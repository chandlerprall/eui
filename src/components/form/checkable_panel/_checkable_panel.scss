.euiPanel.euiCheckablePanel {
  transition: border-color $euiAnimSpeedNormal ease-in;

  & > .euiFlexGroup:first-child .euiCheckablePanel__control {
    border-top-left-radius: $euiBorderRadius;
  }

  & > .euiFlexGroup:last-child .euiCheckablePanel__control {
    border-bottom-left-radius: $euiBorderRadius;
  }
}

.euiPanel.euiCheckablePanel--isChecked {
  border-color: makeHighContrastColor($euiColorPrimary, $euiColorEmptyShade);
}

.euiCheckablePanel__control {
  text-align: center;
  width: $euiSize * 3;
  padding-left: $euiSize;
  padding-right: $euiSize;
  background-color: $euiFormInputGroupLabelBackground;
  transition: background-color $euiAnimSpeedNormal ease-in;

  .euiPanel.euiCheckablePanel--isChecked & {
    background-color: tintOrShade(makeHighContrastColor($euiColorPrimary, $euiColorEmptyShade), 85, 70);
  }
}

.euiCheckablePanel__label {
  padding: $euiSizeS $euiSizeS $euiSizeS $euiSize;
}

.euiCheckablePanel__children {
  padding:  0 $euiSizeS $euiSizeS $euiSize;
}
