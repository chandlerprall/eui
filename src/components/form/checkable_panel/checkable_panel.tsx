import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

import { EuiPanel } from '../../panel';
import { EuiFlexGroup, EuiFlexItem } from '../../flex';
import { EuiRadio, EuiRadioProps } from '../radio';
// @ts-ignore
import { EuiCheckbox } from '../checkbox';
import { EuiFormLabel } from '../form_label';
import { EuiText } from '../../text';

export type EuiCheckablePanelProps = Omit<EuiRadioProps, 'compressed'> & {
  /**
   * Whether the control is a radio button or checkbox
   */
  checkableType?: 'radio' | 'checkbox';
  id: string;
};

export const EuiCheckablePanel: FunctionComponent<EuiCheckablePanelProps> = ({
  children,
  className,
  checkableType = 'radio',
  label,
  ...rest
}) => {
  const { id } = rest;
  const classes = classNames('euiCheckablePanel', className, {
    'euiCheckablePanel--isChecked': rest.checked,
  });

  const CheckableComponent = checkableType === 'radio' ? EuiRadio : EuiCheckbox;

  return (
    <EuiPanel paddingSize="none" className={classes}>
      <EuiFlexGroup
        gutterSize="none"
        alignItems="stretch"
        responsive={false}
        wrap={false}>
        <EuiFlexItem grow={false} className="euiCheckablePanel__control">
          <EuiFlexGroup
            gutterSize="none"
            justifyContent="center"
            alignItems="center">
            <EuiFlexItem>
              <CheckableComponent {...rest} />
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
        <EuiFlexItem className="euiCheckablePanel__label">
          <EuiFormLabel
            htmlFor={id}
            aria-describedby={children ? `${id}-details` : undefined}>
            <EuiText size="m">{label}</EuiText>
          </EuiFormLabel>
        </EuiFlexItem>
      </EuiFlexGroup>
      {children && (
        <EuiFlexGroup gutterSize="none" responsive={false} wrap={false}>
          <EuiFlexItem grow={false} className="euiCheckablePanel__control" />
          <EuiFlexItem
            id={`${id}-details`}
            className="euiCheckablePanel__children">
            {children}
          </EuiFlexItem>
        </EuiFlexGroup>
      )}
    </EuiPanel>
  );
};
