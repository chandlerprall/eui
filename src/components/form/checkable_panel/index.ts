export { EuiCheckablePanel } from './checkable_panel';
