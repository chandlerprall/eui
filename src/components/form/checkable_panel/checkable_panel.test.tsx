import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import { EuiCheckablePanel } from './checkable_panel';

const checkablePanelRequiredProps = {
  id: 'id',
  onChange: () => {},
};

describe('EuiCheckablePanel', () => {
  test('is rendered', () => {
    const component = render(
      <EuiCheckablePanel {...requiredProps} {...checkablePanelRequiredProps} />
    );

    expect(component).toMatchSnapshot();
  });
});
