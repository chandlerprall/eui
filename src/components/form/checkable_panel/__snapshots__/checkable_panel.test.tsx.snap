// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCheckablePanel is rendered 1`] = `
<div
  class="euiPanel euiCheckablePanel testClass1 testClass2"
>
  <div
    class="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero euiCheckablePanel__control"
    >
      <div
        class="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            aria-label="aria-label"
            class="euiRadio euiRadio--noLabel"
            data-test-subj="test subject string"
          >
            <input
              class="euiRadio__input"
              id="id"
              type="radio"
            />
            <div
              class="euiRadio__circle"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiCheckablePanel__label"
    >
      <label
        class="euiFormLabel"
        for="id"
      >
        <div
          class="euiText euiText--medium"
        />
      </label>
    </div>
  </div>
</div>
`;
